src/middlewares/isAccommodationAvail.js

```diff
@@ -0,0 +1,30 @@
+import AccommodationService from '../services/BookingService';
+import Response from '../helpers/Response';
+
+const isAccommodationAvail = async (req, res, next) => {
+  const { checkInDate, roomId } = req.body;
+  const serviceQueries = await AccommodationService.bookAccommodationService(req, res);
+  const {
+    isAccommodationExist, allAvailRoom, isRoomBooked
```

Comment:
I think we can place this middleware under AccommodationMiddlewares as @ivymwende did.
It's because we'll add more middlewares for accommodation.
**PATH**: `src/middlewares/AccommodationMiddleware.js`